scripts/hid_configurator/NrfHidDevice.py

```diff
@@ -377,10 +390,23 @@ def _read_device_info(dev, recipient):
 
         return board_name, hwid
 
+    @staticmethod
+    def _get_peers_cache(dev, recipient):
+        success, fetched_data = NrfHidTransport.exchange_feature_report(dev,
+                                                                   recipient,
+                                                                   0,
+                                                                   ConfigStatus.GET_PEERS_CACHE,
+                                                                   None)
+        if success:
+            return " ".join("0x{:02x}".format(b) for b in fetched_data)
+        else:
+            return None
+
     @staticmethod
     def _get_connected_peers(dev, recipient):
```

Comment:
why do we attach also peers cache results to this function? Can't we have two standalone functions one for getting connected peers and for getting peers cache?

Reply:
I prefer to keep it together (from the end-user perspective cache value is useless standalone).  Later on the `_get_connected_peers` could use peers cache internally to avoid rediscovery if possible (currently we do not maintain list of peers so it's not used).

Reply:
If we add a link between peripherals and dongle (in similar way as it's done in fwupd), we would update it

Reply:
It is not useless as you make many operations in your open_devices function without using the connected peers variable.

And according to cache encoding, you can deduce the number of connected peers without polling the connected peers

Reply:
Yes, the peers cache can be very useful for the **rediscovery**, but it would not be too useful before initial discovery (keep in mind that current implementation of the HID configurator script does not store any information about peers after script finishes execution so peers are fully rediscovered each time script is called by the user).

The only information provided by reading peers cache before the initial discovery is the number of connected peers (keep in mind that a given peer may also "move" in the peripherals array on dongle side - it may be linked with various index of the peers cache). So that actually with the current implementation, the peers cache is mainly read for debug purposes here (there is no dedicated use-case for it at this point). Still I would prefer to keep the functionality testable in scope of NCS.

Reply:
you can keep it testable, I am just talking about two dedicated API calls instead of having one that returns two values

Reply:
If I do it, I would need to get peers cache unconditionally. Current implementation allows me to avoid doing it for peripherals (early return on index peers if functionality is not supported). That helps to avoid assertion failures for peripherals with older firmware. Maybe we could keep it like this? IMHO, it's nothing bad in returning 2 values - especially if they are tightly coupled like in our case.

Reply:
ok